mne/io/eeglab/tests/test_eeglab.py

```diff
@@ -30,17 +30,36 @@
 def test_io_set():
     """Test importing EEGLAB .set files"""
     from scipy import io
-
-    _test_raw_reader(read_raw_eeglab, input_fname=raw_fname, montage=montage)
-    with warnings.catch_warnings(record=True) as w:
+    with warnings.catch_warnings(record=True) as w1:
         warnings.simplefilter('always')
+        # main tests, and test missing event_id
+        _test_raw_reader(read_raw_eeglab, input_fname=raw_fname,
+                         montage=montage)
         _test_raw_reader(read_raw_eeglab, input_fname=raw_fname_onefile,
                          montage=montage)
-        raw = read_raw_eeglab(input_fname=raw_fname_onefile, montage=montage)
-        raw2 = read_raw_eeglab(input_fname=raw_fname, montage=montage)
-        assert_array_equal(raw[:][0], raw2[:][0])
-    # one warning per each preload=False or str with raw_fname_onefile
-    assert_equal(len(w), 3)
+        assert_equal(len(w1), 20)  # for preload_false and dropping events
```

Comment:
A user will get warnings if events are dropped, and events are dropped if they can't be looked up by `event_id_func` or looked up in `event_id`. So if the data only contains parseable events (e.g., 'S100', 'S101'), no warnings will be raised. But for example the example data only contains a bunch of string events, for which warnings will happen unless they're all accounted for by the event_id.